flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/TaskManagerTracker.java

```diff
@@ -21,63 +21,62 @@
 import org.apache.flink.runtime.clusterframework.types.AllocationID;
 import org.apache.flink.runtime.clusterframework.types.ResourceProfile;
 import org.apache.flink.runtime.instance.InstanceID;
-import org.apache.flink.runtime.resourcemanager.WorkerResourceSpec;
 import org.apache.flink.runtime.resourcemanager.registration.TaskExecutorConnection;
-import org.apache.flink.runtime.util.ResourceCounter;
 
-import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Executor;
 
 /** Tracks TaskManager's resource and slot status. */
```

Comment:
We need to describe this class in more detail, similar to the `TaskExecutorManager`.